paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenUI.kt

```diff
@@ -68,45 +74,70 @@ internal fun ManageScreenUI(interactor: ManageScreenInteractor) {
 }
 
 @Composable
-private fun TrailingContent(isSelected: Boolean, isEditing: Boolean, isModifiable: Boolean, paymentMethodId: String?) {
+private fun TrailingContent(
+    isSelected: Boolean,
+    isEditing: Boolean,
+    isModifiable: Boolean,
+    paymentMethod: DisplayableSavedPaymentMethod,
+    interactor: ManageScreenInteractor,
+) {
     if (isEditing && isModifiable) {
         Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
-            EditIcon(paymentMethodId)
-            DeleteIcon(paymentMethodId)
+            EditIcon(paymentMethod)
+            DeleteIcon(paymentMethod, interactor)
         }
     } else if (isEditing) {
-        DeleteIcon(paymentMethodId)
+        DeleteIcon(paymentMethod, interactor)
     } else if (isSelected) {
         SelectedBadge()
     }
 }
 
 @Composable
-private fun DeleteIcon(paymentMethodId: String?) {
+private fun DeleteIcon(paymentMethod: DisplayableSavedPaymentMethod, interactor: ManageScreenInteractor) {
+    val openRemoveDialog = rememberSaveable { mutableStateOf(false) }
+    val paymentMethodId = paymentMethod.paymentMethod.id
+
     TrailingIcon(
         backgroundColor = Color.Red,
         icon = Icons.Filled.Close,
-        modifier = Modifier.testTag("${TEST_TAG_MANAGE_SCREEN_DELETE_ICON}_$paymentMethodId")
+        modifier = Modifier.testTag("${TEST_TAG_MANAGE_SCREEN_DELETE_ICON}_$paymentMethodId"),
+        onClick = {
+            openRemoveDialog.value = true
```

Comment:
We've had issues in the past with missing calls like this. Adding a comment to ensure we have tests covering this.

Reply:
There are two tests in ManageScreenUITest which verify that the dialog opens:
- clickingDeleteIcon_displaysDialog_deletesOnConfirm -- implicitly verifies the dialog is open by clicking the confirm button
- clickingDeleteIcon_displaysDialog_doesNothingOnDismiss -- implicitly verifies the dialog is open by clicking the dismiss button

Just updated the tests to also verify the dialog is hidden after the buttons are clicked.